### DISCUSSION TITLE
Use type of first value as start argument for sum aggregator

### DISCUSSION THREAD

**Discussion by daviewales** [saulpw/visidata#1996] [Ideas]
The `sum` aggregator uses Python's [built-in sum](https://docs.python.org/3/library/functions.html#sum) function:
https://github.com/saulpw/visidata/blob/4ffacbb2ab7b79029bd7304eff0c317e9c009027/visidata/aggregators.py#L128

By default, the `sum` function has the argument `start=0`.
This means that it only works to sum types which are compatible with integers.
For example, if you try to sum `datedelta` values, you get:

    unsupported operand type(s) for +: 'int' and 'datedelta'

Would it make sense to automatically supply the type of the first value we are aggregating to `sum` to allow it to sum types which are not compatible with `int`?

Or is there something else we can add to the definition of `datedelta` to allow it to be summed with the default `start=0`?
https://github.com/saulpw/visidata/blob/4ffacbb2ab7b79029bd7304eff0c317e9c009027/visidata/type_date.py#L96

**Comment by saulpw** [comment]
Sure, that makes sense to me! I committed the change and it seems to work.  (This probably should have been an issue instead.)

**Reply by daviewales** [reply]
Thanks! I wasn't sure where it should live. I can see a 'Create issue from discussion' button in the right sidebar. Do you want me to click it?

**Reply by saulpw** [reply]
Nah, that's okay.  Next time :)
